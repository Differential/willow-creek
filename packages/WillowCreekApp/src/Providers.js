import { nest } from 'recompose';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

export default nest(ClientProvider, Providers);
