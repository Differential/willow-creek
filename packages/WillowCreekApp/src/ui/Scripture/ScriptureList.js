import React from 'react';
import PropTypes from 'prop-types';
import { H4, styled, ButtonLink } from '@apollosproject/ui-kit';

const StyledH4 = styled(({ theme }) => ({
  color: theme.colors.primary,
  textAlign: 'center',
  textDecorationLine: 'underline',
  paddingBottom: theme.sizing.baseUnit * 1.5,
}))(H4);

const ScriptureList = ({
  references,
  onPress,
  tabDestination,
  commas = true,
}) => {
  if (!references && commas) return '';
  if (!references) return [];

  let combo = references.map((reference) => `${reference}`);

  if (commas) {
    combo = combo.join(', ');
  }

  const handleOnPress = () => onPress(tabDestination);

  return (
    <StyledH4>
      <ButtonLink padded onPress={handleOnPress}>
        {combo}
      </ButtonLink>
    </StyledH4>
  );
};

ScriptureList.propTypes = {
  /** Toggles comma formating */
  commas: PropTypes.bool,
  /** The ButtonLink handler */
  onPress: PropTypes.func,
  /** An array of human readable references (i.e. '1 Corinthians 15:57') */
  references: PropTypes.arrayOf(PropTypes.string),
  /** The tab to go to onPress */
  tabDestination: PropTypes.string,
};

export default ScriptureList;
