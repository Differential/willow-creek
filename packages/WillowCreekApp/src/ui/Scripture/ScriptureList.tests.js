import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'WillowCreekApp/src/Providers';

import ScriptureList from './ScriptureList';

const scriptureReferences = ['1 Corinthians 15:57', 'Exodus 17:8-15'];

describe('the ScriptureList component', () => {
  it('renders a scripture List', () => {
    const tree = renderer.create(
      <Providers>
        <ScriptureList onPress={jest.fn()} references={scriptureReferences} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
