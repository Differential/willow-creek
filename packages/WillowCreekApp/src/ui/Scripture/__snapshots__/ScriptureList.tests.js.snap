// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ScriptureList component renders a scripture List 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#00676D",
          "fontFamily": "InterUI-Bold",
          "fontSize": 16,
          "lineHeight": 23.04,
          "paddingBottom": 24,
          "textAlign": "center",
          "textDecorationLine": "underline",
        }
      }
    >
      <Text
        onPress={[Function]}
        padded={true}
        style={
          Object {
            "color": "#17B582",
          }
        }
      >
        1 Corinthians 15:57, Exodus 17:8-15
      </Text>
    </Text>
  </View>
</RCTSafeAreaView>
`;
